lib/systemFlows.js

```diff
@@ -90,6 +109,12 @@ function onConditiononSystemFlowTimeout (callback, args) {
   callback(null, true)
 }
 
+function onConditionSystemNewFirmware (callback, args) {
+  system.getUpdateInfo().then(function (result) {
+    callback(null, (result.length && result[0].version))
```

Comment:
condition callback second params should be boolean.
sth like `callback(null, !!(result.length && result[0].version))`